pymeasure/instruments/hp/hp4145x.py

```diff
@@ -0,0 +1,1252 @@
+#
+# This file is part of the PyMeasure package.
+#
+# Copyright (c) 2013-2025 PyMeasure Developers
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+import time
+import os
+import json
+from enum import IntFlag
+
+import numpy as np
+import pandas as pd
+
+from pymeasure.instruments import Channel, Instrument, SCPIUnknownMixin
+from pymeasure.instruments.validators import (strict_discrete_set,
+                                              truncated_discrete_set,
+                                              strict_range)
+
+import logging
+
+log = logging.getLogger(__name__)
+log.addHandler(logging.NullHandler())
+
+
+######
+# MAIN
+######
+
+
+def check_errors_user_mode_value(cmd_str):
+    data_status = cmd_str[0]
+    channel = cmd_str[1]
+    value = float(cmd_str[4:])
+
+    errors = {
+        'N': "Normal",
+        'L': "Interval is too short",
+        'V': "A-D converter saturation",
+        'X': "Oscillation",
+        'C': "This channel compliance error",
+        'T': "Other channel compliance error"
+    }
+
+    channels = {
+        'A': "SMU1",
+        'B': "SMU2",
+        'C': "SMU3",
+        'D': "SMU4",
+        'E': "VM1",
+        'F': "VM2"
+    }
+
+    if data_status != 'N':
+        log.error(f"HP4145x: Measurement Error - {data_status}, {errors[data_status]} "
+                  f"on channel {channels[channel]}")
+    return value
+
+
+class Status(IntFlag):
+    DATA_READY = 0b00000001
+    SYNTAX_ERROR = 0b00000010
+    END_STATUS = 0b00000100
+    ILLEGAL_PROGRAM = 0b00001000
+    BUSY = 0b00010000
+    SELF_TEST_FAIL = 0b00100000
+    RQS = 0b01000000
+    EMERGENCY = 0b10000000
+    FIXTURE_LID_OPEN = 0b10000010
+    SMU_SHUT_DOWN = 0b10001000
+    POWER_FAILURE = 0b10100000
+
+
+class HP4145x(SCPIUnknownMixin, Instrument):
```

Comment:
Does your device support SCPI commands? If so, please use the SCPIMixin. Otherwise, do not use any mixin.